docs/concepts/what-is-a-failure.md

```diff
@@ -0,0 +1,17 @@
+---
+id: what-is-a-failure
+title: What is a failure?
+sidebar_label: Failure
+description: About failures in Temporal
+tags:
+  - term
+  - explanation
+---
+
+Failure handling is an essential part of development. For more information, including the difference between application-level and platform-level failures, see [Handling Failure From First Principles](https://dominik-tornow.medium.com/handling-failures-from-first-principles-1ed976b1b869). For the practical application of those concepts in Temporal, see [Failure Handling in Practice](https://temporal.io/blog/failure-handling-in-practice).
+
+Temporal Failures are representations (in the SDKs and Event History) of various types of errors that occur in the system.
+
+- [Knowledge Base ▶️ Temporal Failures](/kb/failures)
+
+For languages that throw (or raise) errors (or exceptions), throwing an error that is not a Temporal Failure from a Workflow will fail the Workflow Task (and the Task will be retried until it succeeds), whereas throwing a Temporal Failure will fail the Workflow Execution. For more information, see [Application Failure](/kb/failures#application-failure).
```

Comment:
@cretz how's this?

Reply:
Looks good. Is this in the application development guide? It's an essential part of application development, not a part of "concepts". It's very code/SDK specific. I still think per-lang in the app guide makes more sense.

We have an entire area for the much more obvious "return values" at https://docs.temporal.io/application-development/foundations#workflow-return-values, we might as well have one for errors. Also this doesn't really explain that errors that come from other Temporal calls are also Temporal Failures in this context and just not catching them fails the workflow, but that's probably fine.